test/plugin/test_filter_flowcounter_simple.rb

```diff
@@ -25,9 +25,9 @@ def test_filter
       msgs << {'message'=> 'b' * 100}
     end
     d = create_driver
-    filtered, out = filter(d, msgs)
+    filtered, logs = filter(d, msgs)
     assert_equal msgs, filtered
-    assert { out.include?("count:20") }
+    assert { logs.any? { |log| log.include?("count:20") } }
```

Comment:
Could you show `pp logs`?
I want to think about better assertion with the output.

Reply:
Sure!
For all tests, the number of logs was 1, and it is tab-delimited.

```console
$ bundle exec rake test TESTOPTS='-v'
...
FlowCounterSimpleFilterTest:
  test_filter:
    ["2023-04-05 11:01:34 +0900 [info]: plugin:filter_flowcounter_simple\tcount:20\tindicator:num\tunit:second\n"]
    .: (0.156808)
FlowCounterSimpleOutputTest:
  test_byte:
    ["2023-04-05 11:01:35 +0900 [info]: plugin:out_flowcounter_simple\tcount:3330\tindicator:byte\tunit:second\n"]
    .: (0.133296)
  test_comment:
    ["2023-04-05 11:01:35 +0900 [info]: plugin:out_flowcounter_simple\tcount:3\tindicator:num\tunit:second\tcomment:foobar\n"]
    .: (0.113249)
  test_configure:
    .: (0.007216)
  test_num:
    ["2023-04-05 11:01:35 +0900 [info]: plugin:out_flowcounter_simple\tcount:30\tindicator:num\tunit:second\n"]
    .: (0.111274)
...
```

```diff
diff --git a/test/plugin/test_filter_flowcounter_simple.rb b/test/plugin/test_filter_flowcounter_simple.rb
index aa3ebc1..9635786 100644
--- a/test/plugin/test_filter_flowcounter_simple.rb
+++ b/test/plugin/test_filter_flowcounter_simple.rb
@@ -27,6 +27,7 @@ class FlowCounterSimpleFilterTest < Test::Unit::TestCase
     d = create_driver
     filtered, logs = filter(d, msgs)
     assert_equal msgs, filtered
+    pp logs
     assert { logs.any? { |log| log.include?("count:20") } }
   end

diff --git a/test/plugin/test_out_flowcounter_simple.rb b/test/plugin/test_out_flowcounter_simple.rb
index 2063073..a6fc472 100644
--- a/test/plugin/test_out_flowcounter_simple.rb
+++ b/test/plugin/test_out_flowcounter_simple.rb
@@ -40,6 +40,7 @@ class FlowCounterSimpleOutputTest < Test::Unit::TestCase
       end
       d1.instance.flush_emit(60)
     end
+    pp d1.logs
     assert { d1.logs.any? { |log| log.include?("count:30") } }
   end

@@ -53,6 +54,7 @@ class FlowCounterSimpleOutputTest < Test::Unit::TestCase
       end
       d1.instance.flush_emit(60)
     end
+    pp d1.logs
     assert { d1.logs.any? { |log| log =~ /count:\d+\tindicator:byte\tunit:second/ } }
   end

@@ -66,6 +68,7 @@ class FlowCounterSimpleOutputTest < Test::Unit::TestCase
       end
       d1.instance.flush_emit(60)
     end
+    pp d1.logs
     assert { d1.logs.any? { |log| log.include?("comment:foobar") } }
   end
 end
```

Reply:
How about removing timestamp and using `assert_equal`?

```ruby
def normalize_logs(logs)
  logs.collect do |log|
    log.gsub(/\A\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} [-+]\d{4} \[info\]: /, "") 
  end
end

assert_equal(["plugin:filter_flowcounter_simple\tcount:20\tindicator:num\tunit:second\n"],
             normalize_logs(d1.logs))
```

Reply:
It will show better output (diff of the expected and the actual) on failure.

Reply:
Thanks! Indeed!
I have fixed this.